crates/sui-framework/sources/governance/validator_cap.move

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module sui::validator_cap {
+    use sui::object::{Self, ID, UID};
+    use sui::transfer;
+    use sui::tx_context::TxContext;
+    friend sui::sui_system;
+    friend sui::validator;
+    friend sui::validator_set;
+
+    #[test_only]
+    friend sui::sui_system_tests;
+    #[test_only]
+    friend sui::rewards_distribution_tests;
+
+    /// The capability object is created when creating a new `Validator` or when the
+    /// validator explicitly creates a new capability object for rotation/revocation.
+    /// The holder address of this object can perform some validator operations on behalf of
+    /// the authorizer validator. Thus, if a validator wants to separate the keys for operation
+    /// (such as reference gas price setting or tallying rule reporting) from fund/staking, it
+    /// could transfer this capability object to another address.
+
+    /// To facilitate rotating/revocation, `Validator` stores the ID of currently valid
+    /// `UnverifiedValidatorOperationCap`. Thus, before converting `UnverifiedValidatorOperationCap`
+    /// to `VerifiedValidatorOperationCap`, verification needs to be done to make sure
+    /// the cap object is still valid.
+    struct UnverifiedValidatorOperationCap has key, store {
+        id: UID,
+        authorizer_validator_address: address,
+    }
+
+    /// Privileged operations require `VerifiedValidatorOperationCap` for permission check.
+    /// This is a Hot Potato, only constructed after successful verification,
```

Comment:
Does it need to be a hot potato?  Or could it have `drop`?

Reply:
I think it doesn't have to be hot potato? but it could be good in practice to deconstruct it to use the inner address for verification

Reply:
I think we are good as long as it doesn't have key or store or copy.

Reply:
Agreed, no `key` or `store`.  Omitting `drop` doesn't currently buy you anything, because the underlying address has `drop`, so you would be free to just ignore that after unwrapping.  

What you're describing could be achieved by doing that inner address check in the function that consumes the cap, and that does sound like a neat pattern, as long as the check is the same each time, but looking at the code, it's not, so I don't think there's an easy way to do this.

I'd suggest one of two approaches:

- no `drop`, and add the check that you want to make in the desconstruct function.  Add `copy` to avoid people having to produce multiple verified caps from an unverified one (presumably if you are verified once in a transaction, you are verified for the duration of the transaction).
- Add `drop`, and have functions accept a reference to `VerifiedOperationCap`.

Reply:
If it's a public function, you will not be able to change its signature after the fact, so it will be better keep a consistent signature from the beginning -- given you have `drop` now, it's not going to be a problem if a cap only has one function that would use it, because it will just be dropped at the end of the scope without any extra effort required by the developer.